File: Basic_Bit_Operations/Q1. Number of 1 Bits/Q1_Numberof1Bits.java
class Solution {
    public int hammingWeight(int n) {
        int count = 0; // to store number of 1 bits
        
        // loop until all bits are processed
        while (n != 0) {
            count++;        // found a set bit
            n = n & (n - 1); // remove the rightmost set bit
        }
        
        return count; // return total number of set bits
    }
}

File: Basic_Bit_Operations/Q1. Number of 1 Bits/Q1_Numberof1Bits.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0  # to store number of 1 bits
        
        # loop until all bits are processed
        while n != 0:
            count += 1      # found a set bit
            n = n & (n - 1) # remove the rightmost set bit
        
        return count  # return total number of set bits
